pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -301,6 +303,12 @@ func generateCloudConfig(configFile *ConfigFile) (cloudConfig *CloudConfig, err
 // If no tokenSource is specified, uses oauth2.DefaultTokenSource.
 // If managedZones is nil / empty all zones in the region will be managed.
 func CreateGCECloud(config *CloudConfig) (*GCECloud, error) {
+	// Remove any pre-release version and build metadata from the semver, leaving only the MAJOR.MINOR.PATCH portion.
+        // See http://semver.org/.
+	version := strings.Split(strings.Split(version.Get().GitVersion, "-")[0], "+")[0]
```

Comment:
Isn't `version := strings.Split(v, "-")[0]` sufficient? 

"v1.8.0-rc.1.2+7cc7238996ebf3".  Looks like the + is always on the right side of the -, so only one split is necessary. 
https://github.com/kubernetes/kubernetes/blob/b188868fd929593a0fcf41a7ee2afe3e10b03c39/pkg/version/base.go#L54

Reply:
Reckon the leading `v` be stripped too. Thoughts?

Reply:
My discussions with various people led me to believe that various parts of this can be overridden in some build configuration. I designed this to support the current spec, in which both the pre-release version and build metadata are optional. See: http://semver.org/#spec-item-10.

Reply:
I'll skip the leading 'v'.

Reply:
Okay, sounds good.